dali/ft/ftbase.cpp

```diff
@@ -556,18 +558,21 @@ MemoryBuffer & OutputProgress::deserializeCore(MemoryBuffer & in)
     return in;
 }
 
-MemoryBuffer & OutputProgress::deserializeExtra(MemoryBuffer & in, unsigned version)
+MemoryBuffer & OutputProgress::deserializeExtra(MemoryBuffer & in, unsigned & version)
 {
-    if (in.remaining())
+    if (version>=1 && in.remaining())
     {
-        switch (version)
-        {
-        case 1:
-            in.read(hasCompressed);
-            if (hasCompressed)
-                in.read(compressedPartSize);
-            break;
-        }
+        in.read(hasCompressed);
+        if (hasCompressed && in.remaining())
+            in.read(compressedPartSize);
+    }
+    if (version>=2)
+    {
+        constexpr unsigned bytesRequired=sizeof(numWrites)+sizeof(numReads);
+        if (in.remaining()>=bytesRequired)
+            in.read(numWrites).read(numReads);
```

Comment:
optimization: Could use readPacked (and appendPacked) to shrink the serialization.